### PR TITLE
fix(uploads): add upload.app.box.com as a default upload hostname

### PR DESCRIPTION
**Pull request by satishboggarapu**

### CODE REVIEW

File: src/api/uploads/BaseUpload.js [hunk 1]
```diff
@@ -106,7 +106,7 @@ class BaseUpload extends Base {
 
         const uploadHost = this.getUploadHostFromUrl(upload_url);
         // The default upload host should always be reachable
-        if (uploadHost === `${DEFAULT_HOSTNAME_UPLOAD}/`) {
+        if (uploadHost === `${DEFAULT_HOSTNAME_UPLOAD}/` || uploadHost === `${DEFAULT_HOSTNAME_UPLOAD_APP}/`) {
```
Comment (danubevictoria) [inline src/api/uploads/BaseUpload.js:109]: can we leave a comment here about why we are matching on the 2 default upload hosts?

File: src/constants.js [hunk 2]
```diff
@@ -176,6 +176,7 @@ export const DEFAULT_PATH_STATIC_PREVIEW = 'platform/preview';
 export const DEFAULT_HOSTNAME_API = 'https://api.box.com';
 export const DEFAULT_HOSTNAME_STATIC = 'https://cdn01.boxcdn.net';
 export const DEFAULT_HOSTNAME_UPLOAD = 'https://upload.box.com';
```
Comment (il10) [inline src/constants.js:178]: Can we leave a comment here why we are leaving the const for `https://upload.box.com` and not removing it?

Comment (il10) [reply src/constants.js:178]: ^duplicate of @danubevictoria comment

**Review by il10 — DISMISSED** [review]

**Review by swfree — APPROVED** [review]

**Review by jstoffan — APPROVED** [review]

File: src/api/uploads/BaseUpload.js (hunk 1 above)
Comment (manishPh) [reply src/api/uploads/BaseUpload.js:111]: @satishboggarapu in first line of comment, did you mean to write:
```// DEFAULT_HOSTNAME_UPLOAD is the original default upload host, while DEFAULT_HOSTNAME_UPLOAD_APP is recent.``` ?

Comment (satishboggarapu) [reply src/api/uploads/BaseUpload.js:111]: @manishPh Fixed it.

**Review by manishPh — APPROVED** [review]

**Review by jstoffan — APPROVED** [review]

File: src/api/uploads/__tests__/BaseUpload.test.js [hunk 3]
```diff
@@ -173,6 +173,26 @@ describe('api/uploads/BaseUpload', () => {
             expect(upload.uploadsReachability.isReachable).not.toHaveBeenCalled();
         });
 
+        test('should call preflightSuccessHandler, without doing a reachability test if DEFAULT_HOSTNAME_UPLOAD_APP is provided as uploadUrl', () => {
```
Comment (danubevictoria) [inline src/api/uploads/__tests__/BaseUpload.test.js:176]: i just noticed this, its minor so ok to leave as is, but we can avoid test duplication by using test.each.

File: src/api/uploads/__tests__/UploadsReachability.test.js [hunk 4]
```diff
@@ -100,6 +100,16 @@ describe('api/uploads/UploadsReachability', () => {
             expect(uploadsReachability.getCachedResult).not.toHaveBeenCalled();
             expect(uploadsReachability.makeReachabilityRequest).not.toHaveBeenCalled();
         });
+
+        test('should return true when host is DEFAULT_HOSTNAME_UPLOAD_APP, without checking cache and making a reachability test', async () => {
```
Comment (danubevictoria) [inline src/api/uploads/__tests__/UploadsReachability.test.js:104]: same here test.each

**Pull request merged**